ls1intum/Artemis
# `Programming exercises`: Fix an issue when resetting test cases

### Checklist
#### General



#### Server
#### Client


### Motivation and Context


Accessing the endpoint `api/programming-exercises/{{exerciseId}}/test-cases/reset` currently does not reset visibility but all other settings. Resetting a test case should also reset its visibility to the default configuration (always) 

### Description

Updates `programmingExerciseTestCaseService.reset(exerciseId)` to also reset visibility.
This method is only used by the above mentioned rest endpoint.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise

1. Go to the grading/test case configuration
2. Make some changes to the test cases settings. Do make changes to the visibility so at least one test case does not have visibility == always.
3. Click on the save button and wait for it to finish.
4. Click on the reset button and check that visibility and other options are correctly reset.
